From 034570c6e586f092d43ceafb546fbe0ffe02f012 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Tue, 2 Jul 2024 13:05:59 +0200
Subject: [PATCH] drm/stm: ltdc: remove support of yuv pixel formats for MP21

Soc MP21 &MP 25 doesn't support pixel formats yuv semiplanar & planar
on layer1 only.

Change-Id: I1df57669384b1b0ab220b89a033c899d6681803a
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/390817
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/gpu/drm/stm/ltdc.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/stm/ltdc.c b/drivers/gpu/drm/stm/ltdc.c
index 401915f5e9e8..8ceca74a6680 100644
--- a/drivers/gpu/drm/stm/ltdc.c
+++ b/drivers/gpu/drm/stm/ltdc.c
@@ -1937,10 +1937,11 @@ static struct drm_plane *ltdc_plane_create(struct drm_device *ddev,
 		}
 
 		/*
-		 * Soc MP25 doesn't support pixel formats yuv semiplanar &
+		 * Soc MP21 & MP25 doesn't support pixel formats yuv semiplanar &
 		 * planar on layer1 only.
 		 */
-		if (!(of_device_is_compatible(dev->of_node, "st,stm32mp25-ltdc") && !index)) {
+		if (!((of_device_is_compatible(dev->of_node, "st,stm32mp21-ltdc") ||
+		    of_device_is_compatible(dev->of_node, "st,stm32mp25-ltdc")) && !index)) {
 			if (val & LXCR_C1R_YSPA) {
 				memcpy(&formats[nb_fmt], ltdc_drm_fmt_ycbcr_sp,
 				       ARRAY_SIZE(ltdc_drm_fmt_ycbcr_sp) * sizeof(*formats));
-- 
2.39.5

